### PR TITLE
exEntitySpawner RTTIExtender

### PR DESCRIPTION
**Pull request by Sombra0xCC**
exEntitySpawner

```lua

spawnPosition = Game.GetPlayer():GetWorldTransform()

-- [Simple]
-- Spawn 'Character.Judy' just like prevention system
-- Optionally set appearance to 'judy_diving_suit'
judyEntityID = exEntitySpawner.SpawnRecord('Character.Judy', spawnPosition)
judyEntityID = exEntitySpawner.SpawnRecord('Character.Judy', spawnPosition, 'judy_diving_suit')

-- [Advanced]
-- Spawn base\quest\secondary_characters\judy.ent
-- Optionally set appearance to 'default'
-- Optionally set TweakDB record to 'Character.Judy' (VO/Name/Equipment/etc)
judyEntityID = exEntitySpawner.Spawn([[base\quest\secondary_characters\judy.ent]], spawnPosition)
judyEntityID = exEntitySpawner.Spawn([[base\quest\secondary_characters\judy.ent]], spawnPosition, 'default')
judyEntityID = exEntitySpawner.Spawn([[base\quest\secondary_characters\judy.ent]], spawnPosition, 'default', 'Character.Judy')

-- Some time later...
-- entities are not spawned instantly, FindEntityByID may return nil
judy = Game.FindEntityByID(judyEntityID)

exEntitySpawner.Despawn(judy)
```
You can't mount spawned vehicles for some reasons.

### CODE REVIEW

File: src/overlay/widgets/Console.cpp [hunk 1]
```diff
@@ -149,9 +149,20 @@ void Console::Log(const std::string& acpText)
     std::lock_guard<std::recursive_mutex> _{ m_outputLock };
 
     size_t first = 0;
-    while (first < acpText.size())
+    size_t size = acpText.size();
+    while (first < size)
     {
-        const auto second = acpText.find_first_of('\n', first);
+        // find_first_of \r or \n
+        size_t second = std::string::npos;
```
Comment (WSSDude420) [inline src/overlay/widgets/Console.cpp:156]: should have probably be std::string_view::npos
not like it matters much in the end, but for consistency sake

**Comment by Sombra0xCC** [comment]
> Check failed

Update RED4ext.SDK

**Review by WSSDude420 — COMMENTED** [review]
I believe code looks good, but cannot judge much technicalities around RTTIExtender so...
This is OK by me 👍 

But you should prolly check that RTTIExtender at least afterwards yamashi 😄

**Comment by WSSDude420** [comment]
Also just realized I did not post review till now 🤦

**Comment by WSSDude420** [comment]
If job finishes without issues now, will merge 👍

**Pull request merged**